// @aws-sdk/client-personalize-runtime/dist-es/commands/GetPersonalizedRankingCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPersonalizedRankingRequestFilterSensitiveLog, GetPersonalizedRankingResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetPersonalizedRankingCommand, serializeAws_restJson1GetPersonalizedRankingCommand, } from "../protocols/Aws_restJson1";
export class GetPersonalizedRankingCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, GetPersonalizedRankingCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "PersonalizeRuntimeClient";
        const commandName = "GetPersonalizedRankingCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetPersonalizedRankingRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetPersonalizedRankingResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restJson1GetPersonalizedRankingCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restJson1GetPersonalizedRankingCommand(output, context);
    }
}
